### PR TITLE
Add CFG edges into finally blocks

### LINKED ISSUE #1797: missing CFG edges from return statement to enclosing finally blocks
I notice from [this comment](https://github.com/typetools/checker-framework/blob/4c7fa25c57f251d473d86ff74985d6d240e94e33/dataflow/src/org/checkerframework/dataflow/cfg/CFGBuilder.java#L4151) that there seem to be missing control-flow edges from a `return` statement to the beginning of enclosing finally blocks.  We noticed some unsoundness in NullAway that I think is due to this issue.  E.g., this example gets through NullAway with no warning:
```java
public void foo(@Nullable Object o) {
  try {
    return;
  } finally {
    o.toString();
  }
}
```
Looking under the hood, it seems the dataflow analysis does not even visit the `finally` block, I'm guessing because it starts from the entry and there's no path to the finally block from the entry block in the CFG?

I notice that the above example and other similar examples seem to be checked correctly by the Checker Framework's nullness checker.  Is there some special workaround in that checker that we can borrow?  Is the underlying issue hard to fix?

### PR DESCRIPTION
**Pull request by wmdietl**
As described in #1797 return statements didn't have CFG edges to their enclosing finally blocks.
This actually was a bigger problem, as it also applied to break and continue statements.
This PR makes sure that there are CFG edges *into* finally blocks from all control flow statements.
Some CFG edges *out of* finally blocks still seem incorrect. This will be addressed after #293 in separate PRs. (This has no effect on type checking, but is important for other uses of dataflow.)

Fixes #1797

### CODE REVIEW

**Comment by smillst** [comment]
Could you please change the pull request description to explain how this corrects finally blocks?  I'm not sure what the change is suppose to do, so I can't effectually review it.

**Comment by wmdietl** [comment]
I've added a description to properly explain the PR. I'm not sure how to put that into the title...

**Comment by wmdietl** [comment]
Cross-ref to comment on issue: https://github.com/typetools/checker-framework/issues/1797#issuecomment-363216415

File: dataflow/src/org/checkerframework/dataflow/cfg/CFGBuilder.java [hunk 1]
```diff
@@ -757,6 +781,47 @@ public void popFrame() {
...
+        }
+    }
+
+    /**
+     * A map that keeps track of new labels added within a try block. For names that are outside of
+     * the try block, the finally label is returned. This ensures that a finally block is executed
+     * when control flows outside of the try block.
+     */
+    @SuppressWarnings("serial")
+    protected static class FakeLabelsMap extends HashMap<Name, Label> implements Map<Name, Label> {
```
Comment (wmdietl) [inline dataflow/src/org/checkerframework/dataflow/cfg/CFGBuilder.java]: `FakeLabelsMap` is probably not the best name.
Is `FinallyScopeLabelMap` better? Other suggestions?

Comment (smillst) [reply dataflow/src/org/checkerframework/dataflow/cfg/CFGBuilder.java]: `FinallyScopeLabelMap ` is better.  What about `TryFinallyLablesMap`?   I don't have a preference.

**Pull request merged**